Add the automated utilization reporting doc (#20130)
hashicorp/vault: website/content/api-docs/system/internal-counters.mdx

Any missing parameters are left at their existing value.

<Note>

When [automated utilization
reporting](/vault/docs/enterprise/license/utilization-reporting) is enabled, the

> `retention_months` parameter cannot be configured below the default setting (24)

Reviewer: I think we'll want this to reflect the `minimum_retention_months` setting as well, rather than "24". Maybe something like:

```suggestion
`retention_months` parameter cannot be configured below the default setting ([minimum_retention_months](#read-the-client-count-configuration)).
```

I'm not sure if we want a direct reference here, or if the anchoring is even correct 😅, so I defer to you on the actual change!

> `retention_months` parameter cannot be configured below the default setting ([minimum_retention_months](#read-the-client-count-configuration)).

Reviewer: Oops. Looks like I added a stray `.` here. Also, I didn't realize until checking your deploy links that we can use the https://vault-git-docs-ent-utilization-reporting-hashicorp.vercel.app/vault/api-docs/system/internal-counters#minimum_retention_months anchor directly:
```suggestion
`retention_months` parameter cannot be configured below the default setting ([minimum_retention_months](#minimum_retention_months)
```

## Read the Client Count Configuration

Reading the configuration shows the current settings, as well as a flag as to
whether any data can be queried.

- `enabled` `(string)` - returns `default-enabled` or `default-disabled` if the
  configuration is `default`.
- `queries_available` `(bool)` - indicates whether any usage report is
  available. This will initially be false until the end of the first calendar
  month after the feature is enabled.

> **Vault 1.14.0 or later:** The automated reporting feature exposes two

Reviewer: ```suggestion
**Vault 1.14.0 or later:** The automated reporting feature exposes three
```

additional read-only system settings:

- `reporting_enabled` `(bool)` - indicates whether automated reporting is
  enabled.
- `billing_start_timestamp` `(time)` - returns the configured billing start
  timestamp to be used for automated reporting.
- `minimum_retention_months` `(int)` - returns the configured minimum value for `retention_months`.

<Note>

> The `reporting_enabled` and `billing_start_timestamp` parameters are read-only

Reviewer: This might require some line-wrapping adjustments as well, but:
```suggestion
The `reporting_enabled`,  `billing_start_timestamp`, and `minimum_retention_months` parameters are read-only
```
